multidict/_multidict_base.py

```diff
@@ -1,5 +1,14 @@
+import sys
 from collections.abc import ItemsView, Iterable, KeysView, Set, ValuesView
 
+if sys.version_info >= (3, 11):
+    from typing import assert_never
+else:
+    from typing import Any, NoReturn
+
+    def assert_never(value: Any) -> NoReturn:
+        raise AssertionError(f"Expected code to be unreachable, but got: {value!r}")
```

Comment:
A typical solution is to use `typing_extensions` to shim this. I'll need to be added to the runtime deps in `setup.cfg` with an environment marker, restricting the Python version.